src/H5FDonion.c

```diff
@@ -1182,16 +1182,22 @@ H5FD__onion_open(const char *filename, unsigned flags, hid_t fapl_id, haddr_t ma
 
     /* Copy comment from FAPL info, if one is given */
     if ((H5F_ACC_RDWR | H5F_ACC_CREAT | H5F_ACC_TRUNC) & flags) {
-        if (fa->comment) {
```

Comment:
fa->comment is a fixed-size array, not a pointer to dynamically allocated memory